### ISSUE TITLE
Failing on building docs for 0.10.2

### ISSUE THREAD

**Issue by josephgbr** [Buzztrax/buzztrax#67]
When building docs, I get  the error below. Help please?

> make[3]: Entering directory '/home/rffontenelle/aur/buzztrax/src/buzztrax-0.10.2/docs/reference/bt-core'
  DOC   Preparing build
  DOC   Scanning header files
  DOC   Introspecting gobjects
  DOC   Building XML
  DOC   Building XML
  DOC   Building HTML
Makefile:1170: recipe for target 'html-build.stamp' failed

Environment: Arch Linux, 64-bit, [gtk-doc 1.25-1](https://www.archlinux.org/packages/extra/any/gtk-doc/)
Configure: `./configure --prefix=/usr --enable-coverage --enable-gtk-doc --enable-man --disable-update-icon-cache --disable-update-desktop --disable-update-mime`

Full build log and the aforementioned Makefile in this gist:
https://gist.github.com/josephgbr/b5855fa648aa7cdaf9bc1424b584ae9e

**Comment by ensonic** [comment]
Thanks for the report, could you rerun with '''make V=1''' and paste the log again?

**Comment by josephgbr** [comment]
Sure! Here: https://gist.github.com/josephgbr/a24743c06d17cad2d29fe075f41d85d7

**Comment by ensonic** [comment]
Hmm, just rerun it here and can't reproduce it :/ Unfortunately the logs does not say what actually failed. A few things:
* you pass --enable-coverage, which will make things slower, that's usually only used for special builds to check test coverage, I would not use this for actual packages
* once it fails, if you run make again, does it then work? I wonder if there is a race in the makerules, are you perhaps setting MAKEFLAGS=-jX, where X>1 (this seems to cause some trouble here :/ )

**Comment by ensonic** [comment]
Found it! Thanks for reporting. You can test this my hacking the Makefile.am as in the patch and run automake. Then configure ... as usual.

**Comment by josephgbr** [comment]
I applied the patched, but hadn't much success.

Firstly, my `automake` was updated to version 1.15 and `make` expects version 1.14, complaining as not found: https://gist.github.com/josephgbr/f208db1ec5d4abfb394e236a612bde9d

If I use `autoreconf` before running `configure`, it moves along past `automake` issue, but ends up with same error as before. Please see log: https://gist.github.com/josephgbr/4d05003533f623675bfd44df680b4829

I set MAKEFLAGS="-j1", but nothing changed. (Arch defaults to MAKEFLAGS="-j2", in [makepkg.conf](https://projects.archlinux.org/svntogit/packages.git/tree/trunk?h=packages/pacman))

p.s.: Does my [config.log](https://gist.github.com/josephgbr/b1fe5076a1f2a8273fd8ff831e24fd7a) help?
p.s.2: [Here](https://gist.github.com/josephgbr/57b2e6d785ab01277d6453d4cd2cd60f) is the patch used (extracted with git format-patch)

**Comment by ensonic** [comment]
Thanks a lot for testing! If MAKEFLAGS="-j1" is not fixing it, it is a different issues.  From the logs we see that this chunk fails:
``` bash
rm -rf html && mkdir html && \
mkhtml_options=""; \
gtkdoc-mkhtml 2>&1 --help | grep  >/dev/null "\-\-verbose"; \
if test "$?" = "0"; then \
  if test "x1" = "x1"; then \
    mkhtml_options="$mkhtml_options --verbose"; \
  fi; \
fi; \
gtkdoc-mkhtml 2>&1 --help | grep  >/dev/null "\-\-path"; \
if test "$?" = "0"; then \
  mkhtml_options="$mkhtml_options --path=\"/home/rffontenelle/aur/buzztrax/src/buzztrax-0.10.2/docs/reference/bt-core\""; \
fi; \
cd html && gtkdoc-mkhtml $mkhtml_options  buzztrax-core ../buzztrax-core-docs.xml
Makefile:1182: recipe for target 'html-build.stamp' failed
```

Is there anything created in ```docs/reference/bt-core/html```?

You could also run ```make -k``` to see if that is the only issue.

There is another way, but its a bit tedious. I am the gtk-doc maintainer (by change) and if gtk-doc is build with --enable-debug, one can run GTKDOC_TRACE=1 make 2>&1 >make.log and get lot of outputs. This could help to see where it fails. Ideally gtk-doc would tell us.

**Comment by josephgbr** [comment]
> Is there anything created in docs/reference/bt-core/html?

Yep, there is a CSS and also some HTML and PNG files.

> You could also run make -k to see if that is the only issue.

Here is the output: https://gist.github.com/josephgbr/bd1cbb8f7c94b9cff0d272e0d080c782

> There is another way, but its a bit tedious. I am the gtk-doc maintainer (by change) and if gtk-doc is build with --enable-debug, one can run GTKDOC_TRACE=1 make 2>&1 >make.log and get lot of outputs. This could help to see where it fails. Ideally gtk-doc would tell us.

gtk-doc compilation failed because it required "DocBook XML DTD V4.3 in XML catalog", but I had [docbook-xml-dtd version 4.4](https://aur.archlinux.org/packages/docbook-xml-dtd/).  Unisntalling docbook-xml-dtd didn't solved either.  Do I really need docbook-xml-dtd? If not, how can I disable this requirement in gtk-doc's `configure` options?

**Comment by josephgbr** [comment]
Tried another way: I disabled doc building by removing `--enable-gtk-doc` and adding `make check`. The result was the error below. See [full content of test-suite.log](https://gist.github.com/josephgbr/3df7a56cda7fc49f38ca1f955a3b9c0e).

> I/O error : Attempt to load network entity http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd
./docs/help/bt-edit/C/buzztrax-edit.xml:9: warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd"
]>
  ^
I/O error : Attempt to load network entity http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd
warning: failed to load external entity "http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd"
validity error : Could not load the external subset "http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd"
Document ./docs/help/bt-edit/C/buzztrax-edit.xml does not validate

Could this be related?

**Comment by ensonic** [comment]
Hi,

I think you don't have the docbook dtd installed, I'll see if I can detect that and skip the tests. On my machine I have e.g.:
/usr/share/sgml/docbook/dtd/4.2/docbookx.dtd
/usr/share/sgml/docbook/dtd/4.3/docbookx.dtd
/usr/share/sgml/docbook/dtd/4.4/docbookx.dtd
/usr/share/sgml/docbook/dtd/4.5/docbookx.dtd

**Comment by josephgbr** [comment]
I finally figured out what was wrong, and it was on my side: It was a badly maintained package from AUR (i.e. platform with unsupported packages provided by users, therefore they can be dangerous) which overwrites the XML catalog that provides location of the docbook DTD files, providing an incomplete one. I installed this terrible package in April 11 probably thinking that it would provide the DTD required - which I was wrong. After removing it and reinstalling the appropriate package (that would be [docbook-xml](https://www.archlinux.org/packages/extra/any/docbook-xml/)), compilation with `--enable-gtk-doc` worked just fine! Sorry for the inconvenience.
